Shopify/cli
# Requested scopes not fetched from API Client configuration

### WHY are these changes introduced?
Versioned app configuration has been 100% rolled out altogether with the 3.55.0 release. That means that the CLI should not use `api client configuration` anymore



### WHAT is this pull request doing?
- Remove `requestedAccessScopes` from any `app` API query
- Use `scopes` from the current app version configuration where required


### How to test your changes?
- `deploy` the app with an specific set of `scopes`
- Modify the `scopes` inside the `toml` and run the `dev` command. You should see the warn banner saying the remote and local scopes are different. The remote scopes should match the ones `deployed` previously


### Post-release steps



### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
